monkey/tests/unit_tests/monkey_island/cc/repositories/test_plugin_archive_parser.py

```diff
@@ -32,6 +48,148 @@ def dir_tarfile(dir_plugin_file) -> TarFile:
     return tarfile.open(dir_plugin_file)
 
 
+@pytest.fixture
+def plugin_with_three_vendors_tarfile(plugin_with_three_vendors_file) -> TarFile:
+    return tarfile.open(plugin_with_three_vendors_file)
+
+
+@pytest.fixture
+def plugin_with_two_vendor_dirs_one_vendor_file_tarfile(
+    plugin_with_two_vendor_dirs_one_vendor_file_file,
+) -> TarFile:
+    return tarfile.open(plugin_with_two_vendor_dirs_one_vendor_file_file)
+
+
+@pytest.fixture
+def only_windows_vendor_plugin_tarfile(only_windows_vendor_plugin_file) -> TarFile:
+    return tarfile.open(only_windows_vendor_plugin_file)
+
+
+def _get_plugin_source_tar(tarfile_):
+    return TarFile(fileobj=io.BytesIO(get_plugin_source(tarfile_)))
+
+
+def test_get_os_specific_plugin_source_archives(two_vendor_plugin_tarfile, tmp_path):
+    plugin_source_tar = _get_plugin_source_tar(two_vendor_plugin_tarfile)
+    plugin_vendors = get_plugin_source_vendors(plugin_source_tar)
+    os_specific_data = get_os_specific_plugin_source_archives(
+        plugin_source_tar, plugin_vendors, tmp_path
+    )
+
+    assert os_specific_data[OperatingSystem.WINDOWS] != os_specific_data[OperatingSystem.LINUX]
+
+    assert not os.path.exists(tmp_path / "linux")
+    assert not os.path.exists(tmp_path / "windows")
```

Comment:
Unit tests should be focused around `parse_plugin()`. I push a tarfile in and I validate the output.

Reply:
There are tests for `parse_plugin()` as well.